src/routes/formsgSiteLaunch.ts

```diff
@@ -162,16 +169,26 @@ export class FormsgSiteLaunchRouter {
     // 1. Extract arguments
     const { submissionId } = req.body.data
     const { responses } = res.locals.submission
-    const formResponses: FormResponsesProps = {
-      submissionId,
-      requesterEmail: getField(responses, REQUESTER_EMAIL_FIELD),
-      repoName: getField(responses, REPO_NAME_FIELD),
-      primaryDomain: getField(responses, PRIMARY_DOMAIN),
-      redirectionDomain: getField(responses, REDIRECTION_DOMAIN),
-      agencyEmail: getField(responses, AGENCY_EMAIL_FIELD),
-    }
+    const siteLaunchList = getFieldsFromTable(responses, SITE_LAUNCH_LIST)
+    const formResponses: FormResponsesProps[] = []
+    siteLaunchList?.forEach((element) => {
+      // todo some sort of input validation here for table values
+
+      const formResponse: FormResponsesProps = {
+        submissionId,
+        requesterEmail: getField(responses, REQUESTER_EMAIL_FIELD),
+        repoName: element[2],
+        primaryDomain: element[0],
+        redirectionDomain: element[1] === "WWW" ? `www.${element[0]}` : "",
+        // if agency email not needed, use email from requester instead
+        agencyEmail: element[3]
+          ? element[3]
+          : getField(responses, REQUESTER_EMAIL_FIELD),
+      }
```

Comment:
i think keying by index here for element is rather confusing (not your fault); could we add a comment elaborating on what the indexes of the element map to?

Reply:
4b9cc76